cluster/aws/util.sh

```diff
@@ -721,14 +723,11 @@ function create-iam-profile {
 
 # Creates the IAM roles (if they do not already exist)
 function ensure-iam-profiles {
-  aws iam get-instance-profile --instance-profile-name ${IAM_PROFILE_MASTER} || {
```

Comment:
> All these examples are within the e2e framework? 

No and so does this file. This file isn't used only in our test framework. There are real users, using `kube-up` in production today. And they could be using it for multiple clusters.

> If you look at any other type of AWS resource which is created in kube-up.sh, it's also destroyed in kube-down.sh.

Yeah, but this isn't deleting in `kube-down.sh`. So it is not exactly deleting what it created. It is reversing the model. It might accidentally delete what it did not create when it comes up the first time. As far as I see, it is deleting in `kube-up.sh`. Am I wrong/missing something here?

> If we don't do this, then any changes made to the IAM roles/instance profiles will not be propagate properly to e2e tests.

This smells like something that must be done in an e2e specific environment, not here, given that users are going to use in their production environments.

> Finally, why are we so concerned about deleted these IAM roles? They're created and used exclusively by the cluster (just like all the other AWS resources),

Yeah, but that assumes that there is only one cluster. There could another cluster with the same name. We can't simply go ahead and delete its IAM roles without any warning. 

> can you state explicitly the rationale behind not deleting them when the cluster is torn down?

The rationale is above. But stating it again. We can't delete a resource, as part of the bootstrap, unless we are 100% sure we created them. There is nothing here to guarantee that. It is not even close. We are deleting without any checks, without any warnings in place.